api_v3/services/UserAppRoleService.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * Manage application based roles for user
+ *
+ * @service userAppRole
+ * @package api
+ * @subpackage services
+ */
+class UserAppRoleService extends KalturaBaseService
+{
+	/**
+	 * @param $serviceId
+	 * @param $serviceName
+	 * @param $actionName
+	 * @return void
+	 * @throws KalturaAPIException
+	 */
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		
+		// Add Partner Ids to criteria for the following peers during execution of that service
+		$this->applyPartnerFilterForClass('UserRole');
```

Comment:
Double-check other object that you access as part of this service if you need to apply the filter on them as well

Reply:
No need
We call it in parent::initService at:
https://github.com/kaltura/server/blob/cc0a987b70fc98d4620f2f3ab73d8c0ae1910e45/api_v3/lib/KalturaBaseService.php#L176

which will get to
https://github.com/kaltura/server/blob/cc0a987b70fc98d4620f2f3ab73d8c0ae1910e45/alpha/apps/kaltura/lib/myPartnerUtils.class.php#L181-L180

Where we have 'kuser' included